WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -817,4 +818,26 @@ def create
     end
     params.require(:registration).permit(*permitted_params)
   end
+
+  private def registration_from_params
+    id = params.require(:id)
+    Registration.find(id)
+  end
+
+  def create_paypal_order
+    unless Rails.env.production?
```

Comment:
See remark above. In these pure, JSON-based APIs, you don't have to show a warning message. You can just do

```ruby
def create_paypal_order
  if Rails.env.production?
    return head :forbidden # sends a 403 to the client
  end
  
  # ... do whatever you want to do...
end
```